### PR TITLE
fix(dialog): avoid subpixel rendering issues and refactor GlobalPositionStrategy

### LINKED ISSUE #932: bug(overlay): overlays are sometimes blurred in Chrome
I think you should change the text to "Use your eye glasses to see tooltips"

image

PS: My chrome zoom is set to 100%

### PR DESCRIPTION
**Pull request by crisbeto**
Refactors the `GlobalPositionStrategy` to use flexbox for positioning. This avoids the subpixel rendering issues that come with using transforms for centering.

Fixes #932.

### CODE REVIEW

File: src/lib/core/overlay/_overlay.scss [hunk 1]
```diff
@@ -20,9 +17,21 @@
...
     width: 100%;
+  }
+
+  // The overlay-container is an invisible element which contains all individual overlays.
+  .md-overlay-container {
+    position: fixed;
     z-index: $md-z-index-overlay-container;
   }
 
+  // Wrapper element for the panes that use the GlobalPositionStrategy.
```
Comment (jelbourn) [inline src/lib/core/overlay/_overlay.scss]: Expand this with something like
```ts
// We use an extra wrapper element in order to use make the overlay itself a flex item.
// This makes centering the overlay easy without running into the subpixel rendering
// problems tied to using `transform`.
```

File: src/lib/core/overlay/position/global-position-strategy.ts [hunk 2]
```diff
@@ -1,87 +1,71 @@
...
-    return this;
-  }
+  private _cssPosition: string = 'static';
+  private _topOffset: string = '';
+  private _bottomOffset: string = '';
+  private _leftOffset: string = '';
+  private _rightOffset: string = '';
+  private _alignItems: string = '';
+  private _justifyContent: string = '';
+  private _wrapper: HTMLElement;
```
Comment (jelbourn) [inline src/lib/core/overlay/position/global-position-strategy.ts]: Add description for `_wrapper` that explains what it's for and that it's lazily created.

File: src/lib/core/overlay/position/position-strategy.ts [hunk 3]
```diff
@@ -3,4 +3,7 @@ export interface PositionStrategy {
 
   /** Updates the position of the overlay element. */
   apply(element: Element): Promise<void>;
+
+  /** Cleans up any DOM modifications made by the position strategy, if necessary. */
+  dispose?(): void;
```
Comment (jelbourn) [inline src/lib/core/overlay/position/position-strategy.ts]: Let's make this required and give `ConnectedPositionStrategy` a no-op `dispose`, then we won't have to check whether it exists before calling it.

**Comment by crisbeto** [comment]
Applied the feedback and rebased.

**Comment by jelbourn** [comment]
LGTM

**Pull request merged**